lib/jira.py

```diff
@@ -65,31 +97,34 @@ def create_json_record(issues, url):
     data = {
       "releaseNotes": {
          "issues": {
-            "fixed": 
+            "fixed":
                 [{ "id": issue, "source": url }  for issue in issues]
          }
       }
-    }                         
-    #return json.dumps(data)
-    return data
-    
-
-def parse_credentials_file(credentials_file):
-    """
-    format credentials file:
-    {
-         "api_token": "token_id"
     }
-    """
-    # Open and read the JSON file
-    with open(credentials_file, 'r') as file:
-        data = json.load(file)
     return data
 
 
-def query_jira(jira_domain_url, jql_query, credentials_file):
-    credentials = parse_credentials_file(credentials_file)
-    api_token = credentials["api_token"]
+def get_secret_data(namespace, secret_name):
+    log(f"secret_name: {secret_name}")
+    cmd = ["kubectl", "get", "secret", secret_name, "-n", namespace, "-ojson"]
+    try:
+        cmd_str = " ".join(cmd)
+        log(f"Running {cmd_str}")
+        result = subprocess.run(cmd, check=True, capture_output=True, text=True)
+    except subprocess.CalledProcessError:
+        log(f"Command {cmd_str} failed, check exception for details")
```

Comment:
Since the command string will contain spaces, you should delineate the text so it is clear what is description and what is command line.

```suggestion
        logging.warning(f"Command '{cmd_str}' failed, check exception for details")
```

Reply:
fixed